cpp/daal/src/externals/service_service_mkl.h

```diff
@@ -37,49 +37,106 @@ namespace mkl
 {
 struct MklService
 {
-    static void * serv_malloc(size_t size, size_t alignment) { return fpk_serv_malloc(size, alignment); }
+    static void * serv_malloc(size_t size, size_t alignment) { return MKL_malloc(size, alignment); }
 
-    static void serv_free(void * ptr) { fpk_serv_free(ptr); }
+    static void serv_free(void * ptr) { MKL_free(ptr); }
 
-    static void serv_free_buffers() { fpk_serv_free_buffers(); }
+    static void serv_free_buffers() { MKL_Free_Buffers(); }
 
     static int serv_memcpy_s(void * dest, size_t destSize, const void * src, size_t srcSize)
     {
-        return fpk_serv_memcpy_s(dest, destSize, src, srcSize);
+        if (destSize < srcSize) return static_cast<int>(ENOMEM);
+        memcpy(dest, src, srcSize);
```

Comment:
`memcpy_s` and other safe functions are available in C++11. Can you try to use those?

Reply:
I was trying but it returns undefined_references to these functions even with different headers. Looks like can be added later